File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        boolean flag=false;
        int a[]=new int[2];
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i]) && map.get(target-nums[i])!=i)
            {
                a[0]=i;
                a[1]=map.get(target-nums[i]);
                break;
            }
        }
        return a;
        
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        vector<int> v;
        for(int i=0;i<nums.size();i++){
            mp.insert({nums[i],i});
        }
        for(int i=0;i<nums.size();i++){
            if(mp.find(target-nums[i])!=mp.end() && i!=mp.find(target-nums[i])->second){
                v.push_back(i);
                v.push_back(mp.find(target-nums[i])->second);
                break;
            }
        }
        return v;
    }
};